### PR TITLE
#50 create job posting

### PR DESCRIPTION
**Pull request by fullKTRL**
Created JobPosting.java
Created JobType.java

### CODE REVIEW

File: synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package com.soen.synapsis.appuser.jobposting;
+
+import com.soen.synapsis.appuser.AppUser;
+import com.soen.synapsis.appuser.Role;
+
+public class JobPosting {
```
Comment (JamesPartsafas) [inline synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java:6]: Is this class meant to represent a data model? If so, it should be marked with @Entity, and the attributes given appropriate annotations as well for JPA to work (see AppUser as an example of what I mean).

File: synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    private final AppUser creator;
```
Comment (JamesPartsafas) [inline synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java:8]: Should this just be an ID foreign key to the appUser that created the posting?

File: synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    private String position;
+    private String company;
```
Comment (JamesPartsafas) [inline synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java:10]: This as well I think should be an ID foreign key to the company (and don't forget that companies are just appUsers with a company role).

File: synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    private String address;
+    private String description;
+    private JobType type;
+    private int numAvailable;
+    private int numApplicants;
+
+    public JobPosting (AppUser creator, String position, String company, String address, String description, JobType type, int numAvailable, int numApplicants) {
+        if (creator.getRole() != Role.RECRUITER) {
```
Comment (JamesPartsafas) [inline synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java:18]: I don't think this sort of business logic should be in the data model class. It seems more like something that belongs in a service layer.

**Comment by JamesPartsafas** [comment]
I think for posting jobs, we'll need a new JobController as well as a JobService to handle all the relevant business logic, as well as a form on the frontend to add the data. As it is, this PR contains only the data model class. Are the other components going to be in different PRs?

File: synapsis/src/test/java/com/soen/synapsis/unit/appuser/jobposting/JobPostingTest.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import com.soen.synapsis.appuser.Role;
+import com.soen.synapsis.appuser.jobposting.JobPosting;
+import com.soen.synapsis.appuser.jobposting.JobType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+class JobPostingTest {
```
Comment (JamesPartsafas) [inline synapsis/src/test/java/com/soen/synapsis/unit/appuser/jobposting/JobPostingTest.java:14]: The tests look good.

**Comment by fullKTRL** [comment]
> I think for posting jobs, we'll need a new JobController as well as a JobService to handle all the relevant business logic, as well as a form on the frontend to input the data. As it is, this PR contains only the data model class. Are the other components going to be in different PRs?

Yes this pull request is only for the data model

**Review by JamesPartsafas — APPROVED** [review]
Looks well done.

File: synapsis/src/main/resources/templates/pages/createJob.html [hunk 6]
```diff
@@ -0,0 +1,81 @@
+<!DOCTYPE html>
```
Comment (CHADBUCK13) [inline synapsis/src/main/resources/templates/pages/createJob.html:1]: Why does this html file have a header? Does it not exist as a fragment within the layout?

File: synapsis/src/main/resources/templates/pages/job.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
Comment (CHADBUCK13) [inline synapsis/src/main/resources/templates/pages/job.html:1]: Same here

**Pull request merged**